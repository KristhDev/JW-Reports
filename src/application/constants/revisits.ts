export const revisitsMessages = {
    ABOUT_MIN_LENGTH: 'La información de la persona debe tener al menos 10 caracteres.',
    ABOUT_REQUIRED: 'La información de la persona es requerida.',
    ADDED_SUCCESS: 'Ha agregado la revisita correctamente.',
    ADDRESS_MIN_LENGTH: 'La dirección debe tener al menos 10 caracteres.',
    ADDRESS_REQUIRED: 'La dirección es requerida.',
    COMPLETED_SUCCESS: 'Ha marcado como completa su revisita correctamente.',
    DELETED_SUCCESS: 'Ha eliminado la revisita correctamente.',
    EXPORTED_SUCCESS: 'Ha exportado sus revisitas correctamente. El archivo se encuentra en la carpeta de descargas de su dispositivo.',
    NEXT_VISIT_REQUIRED: 'La fecha de la último visita no puede estar vacía.',
    PERSON_MIN_LENGTH: 'El nombre de la persona debe tener al menos 2 caracteres.',
    PERSON_NAME_REQUIRED: 'El nombre de la persona es requerido.',
    UNSELECTED_COMPLETE: 'No hay una revisita seleccionada para completar.',
    UNSELECTED_DELETE: 'No hay una revisita seleccionada para eliminar.',
    UNSELECTED_UPDATE: 'No hay una revisita seleccionada para actualizar.',
    UPDATED_SUCCESS: 'Ha actualizado la revisita correctamente.',
}