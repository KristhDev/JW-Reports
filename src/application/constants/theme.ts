import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];