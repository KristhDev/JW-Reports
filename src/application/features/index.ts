export * from './auth';
export * from './courses';
export * from './lessons';
export * from './permissions';
export * from './preaching';
export * from './revisits';
export * from './status';
export * from './types';
export * from './ui';