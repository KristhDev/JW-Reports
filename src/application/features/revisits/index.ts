export {
    default as revisitsReducer,
    addRevisit,
    addRevisits,
    clearRevisits,
    INIT_REVISIT,
    removeRevisit,
    removeRevisits,
    REVISITS_INITIAL_STATE,
    setHasMoreRevisits,
    setIsLastRevisitLoading,
    setIsRevisitDeleting,
    setIsRevisitsExporting,
    setIsRevisitLoading,
    setIsRevisitsLoading,
    setLastRevisit,
    setRefreshRevisits,
    setRevisitFilter,
    setRevisits,
    setRevisitsPagination,
    setRevisitsScreenHistory,
    setSelectedRevisit,
    updateRevisit
} from './slice';

export * from './types';