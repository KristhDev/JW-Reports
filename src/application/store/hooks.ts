import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();