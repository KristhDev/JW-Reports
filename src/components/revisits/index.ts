export { RevisitCard } from './RevisitCard';
export { RevisitForm } from './RevisitForm';