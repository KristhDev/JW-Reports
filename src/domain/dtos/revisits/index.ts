export * from './complete.dto';
export * from './create.dto';
export * from './update.dto';