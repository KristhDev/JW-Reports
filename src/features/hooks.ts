import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;