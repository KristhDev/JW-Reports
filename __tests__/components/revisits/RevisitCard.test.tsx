import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { MenuProvider } from 'react-native-popup-menu';
import dayjs from 'dayjs';

/* Features */
import { selectedRevisitState } from '../../features/revisits';

/* Components */
import { RevisitCard } from '../../../src/components/revisits';

/* Hooks */
import { useRevisits, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Setup */
import { navigateMock } from '../../../jest.setup';

const setSelectedRevisitMock = jest.fn();
const onDeleteMock = jest.fn();
const onPassMock = jest.fn();
const onRevisitMock = jest.fn();

const selectedRevisit = selectedRevisitState.selectedRevisit;

/* Mock hooks */
jest.mock('../../../src/hooks/useRevisits.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <RevisitCard /> component', () => {
    (useRevisits as jest.Mock).mockReturnValue({
        setSelectedRevisit: setSelectedRevisitMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
        BUTTON_TRANSPARENT_COLOR: 'rgba(255, 255, 255, 0.15)'
    });

    beforeEach(() => {
        render(
            <MenuProvider>
                <RevisitCard
                    onDelete={ onDeleteMock }
                    onPass={ onPassMock }
                    onRevisit={ onRevisitMock }
                    revisit={ selectedRevisit }
                />
            </MenuProvider>
        );
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render data of revisit', () => {
        const nextVisit = dayjs(selectedRevisit.next_visit);
        const visit = `Visitar el ${ nextVisit.format('DD') } de ${ nextVisit.format('MMMM') } del ${ nextVisit.format('YYYY') }`;

        /* Get elements with data of revisit */
        const nextVisitText = screen.getByTestId('revisit-card-next-visit-text');
        const personNameText = screen.getByTestId('revisit-card-person-name-text');
        const aboutText = screen.getByTestId('revisit-card-about-text');

        /* Check if elements exists and contain data of revisit */
        expect(nextVisitText).toBeTruthy();
        expect(nextVisitText.props.children).toBe(visit);
        expect(personNameText).toBeTruthy();
        expect(personNameText.props.children).toBe(selectedRevisit.person_name);
        expect(aboutText).toBeTruthy();
    });

    it('should call setSelectedRevisit and navigate when card is pressed', () => {

        /* Get touchable of card */
        const touchable = screen.getByTestId('revisit-card-touchable');
        fireEvent.press(touchable);

        /**
         * Check setSelectedRevisit and navigate are called on time
         * with respective args
         */
        expect(setSelectedRevisitMock).toHaveBeenCalledTimes(1);
        expect(setSelectedRevisitMock).toHaveBeenCalledWith(selectedRevisit);
        expect(navigateMock).toHaveBeenCalledTimes(1);
        expect(navigateMock).toHaveBeenCalledWith('RevisitDetailScreen');
    });

    it('should render Visita hecha when selectedRevisit.done is true', () => {
        render(
            <MenuProvider>
                <RevisitCard
                    onDelete={ onDeleteMock }
                    onPass={ onPassMock }
                    onRevisit={ onRevisitMock }
                    revisit={{ ...selectedRevisit, done: true }}
                />
            </MenuProvider>
        );

        /* Get visit text */
        const nextVisitText = screen.getByTestId('revisit-card-next-visit-text');

        /* Check if element exists and contain text "Visita hecha" */
        expect(nextVisitText).toBeTruthy();
        expect(nextVisitText.props.children).toBe('Visita hecha');
    });
});