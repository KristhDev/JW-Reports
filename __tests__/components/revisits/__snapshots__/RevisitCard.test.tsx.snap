// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RevisitCard /> component should to match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          false,
          {
            "marginVertical": 8,
            "width": 734,
          },
        ]
      }
      testID="revisit-card-touchable"
    >
      <View
        style={
          {
            "backgroundColor": "#292929",
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#BFBFBF",
              "fontSize": 14,
              "marginBottom": 15,
            }
          }
          testID="revisit-card-next-visit-text"
        >
          Visitar el 09 de December del 2021
        </Text>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
          testID="revisit-card-person-name-text"
        >
          Dariana Hand
        </Text>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
            }
          }
          testID="revisit-card-about-text"
        >
          Ratione quae debitis pariatur tempora magnam est voluptatibus. Eveniet ipsum omnis maxime et soluta iusto. Autem aut ducimus dolore et recusandae et inventore. Enim autem in hic optio.
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 50,
                  "height": 42,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 42,
                },
                {
                  "height": 35,
                  "position": "absolute",
                  "right": -2.5,
                  "top": 5,
                  "width": 35,
                },
              ],
            ]
          }
          testID="fab-touchable"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#C0A7E1",
                  "fontSize": 20,
                },
                {
                  "marginLeft": 2,
                },
                {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ïŠ¤
          </Text>
        </View>
        <View
          style={
            {
              "position": "absolute",
              "right": 20,
              "top": 30,
            }
          }
        >
          <View
            collapsable={false}
          >
            <View
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                ctx={
                  {
                    "menuActions": {
                      "_getOpenedMenu": [Function],
                      "_notify": [Function],
                      "closeMenu": [Function],
                      "isMenuOpen": [Function],
                      "openMenu": [Function],
                      "toggleMenu": [Function],
                    },
                    "menuRegistry": {
                      "getAll": [Function],
                      "getMenu": [Function],
                      "setOptionsCustomStyles": [Function],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                      "updateLayoutInfo": [Function],
                    },
                  }
                }
                style={
                  [
                    undefined,
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
